client/src/components/Form/Elements/FormSelect.vue

```diff
@@ -0,0 +1,182 @@
+<script setup lang="ts">
+import { computed, onMounted, watch, type ComputedRef } from "vue";
+import Multiselect from "vue-multiselect";
+
+type SelectValue = string | null;
+
+interface SelectOption {
+    label: string;
+    value: SelectValue;
+}
+
+const props = withDefaults(
+    defineProps<{
+        multiple?: boolean;
+        optional?: boolean;
+        options: Array<[string, string]>;
+        value?: Array<string> | string;
+    }>(),
+    {
+        multiple: false,
+        optional: false,
+        value: null,
+    }
+);
+
+const emit = defineEmits<{
+    (e: "input", value: SelectValue | Array<SelectValue>): void;
+}>();
+
+/**
+ * Determine dom wrapper class
+ */
+const cls: ComputedRef<string> = computed(() => {
+    return props.multiple ? "form-select-multiple" : "form-select-single";
+});
+
+/**
+ * Configure deselect label
+ */
+const deselectLabel: ComputedRef<string> = computed(() => {
+    return props.multiple ? "Click to remove" : "";
+});
+
+/**
+ * Translates input options for consumption by the
+ * select component into an array of objects
+ */
+const formattedOptions: ComputedRef<Array<SelectOption>> = computed(() => {
+    const result: Array<SelectOption> = props.options.map((option: [string, string]) => ({
+        label: option[0],
+        value: option[1],
+    }));
+    if (props.optional && !props.multiple) {
+        result.unshift({
+            label: "Nothing selected",
+            value: null,
+        });
+    }
+    return result;
+});
+
+/**
+ * Tracks if the select field has options
+ */
+const hasOptions: ComputedRef<Boolean> = computed(() => {
+    return formattedOptions.value.length > 0;
```

Comment:
Is that correct for optional non-multi options where you inject null ?

Reply:
Yes, since options is always an array and as such formattedOptions are always an array too.